app/src/main/java/com/cornellappdev/uplift/ui/MainNavigationWrapper.kt

```diff
@@ -70,136 +79,162 @@ fun MainNavigationWrapper(
-                        classesViewModel = classesViewModel
-                    )
+            composable<UpliftRootRoute.GymDetail> {
+                GymDetailScreen(
+                    gymDetailViewModel = gymDetailViewModel,
+                    navController = navController,
+                    classDetailViewModel = classDetailViewModel
+                ) {
+                    navController.popBackGym(gymDetailViewModel)
                 }
-                composable(route = "classDetailClasses") {
-                    ClassDetailScreen(
-                        classDetailViewModel = classDetailViewModel,
-                        navController = navController
-                    ) {
-                        navController.popBackClass(classDetailViewModel)
-                    }
+            }
+            // TODO: I split these across multiple screens to make it so the user sticks to
+            //  the side of the app they were originally on.
+            //  However, I think this might cause a bug if you have classes open on BOTH
+            //  sides of the app then pop back. I think both will end up popping back.
+            //  I can't test RN cuz backend is down, so test this and see. If it's broken,
+            //  change it to just use one "classDetail" route on the class half.
+            composable<UpliftRootRoute.ClassDetail> {
+                ClassDetailScreen(
+                    classDetailViewModel = classDetailViewModel, navController = navController
+                ) {
+                    navController.popBackClass(classDetailViewModel)
                 }
             }
-            navigation(startDestination = "sportsMain", route = "sports") {
-                composable(route = "sportsMain") {}
+            composable<UpliftRootRoute.Classes> {
+                ClassScreen(
+                    classDetailViewModel = classDetailViewModel,
+                    navController = navController,
+                    classesViewModel = classesViewModel
+                )
+            }
+            composable<UpliftRootRoute.ReportIssue> {
+                ReportIssueScreen(
+                    onSubmit = reportViewModel::createReport,
+                    onBack = { navController.popBackStack() }
+                )
             }
-            navigation(startDestination = "favoritesMain", route = "favorites") {
-                composable(route = "favoritesMain") {}
+            composable<UpliftRootRoute.ReportSuccess> {
+                ReportSubmittedScreen(
+                    onSubmitAnother = reportViewModel::navigateToReport,
+                    onReturnHome = reportViewModel::navigateToHome
+                )
             }
+            composable<UpliftRootRoute.Sports> {}
+            composable<UpliftRootRoute.Favorites> {}
         }
     }
 }
+
+@Serializable
+sealed class UpliftRootRoute {
+    @Serializable
+    data object Home : UpliftRootRoute()
+
```

Comment:
Lovely stuff, lots of migrating work you didn't have to do that makes this codebase a LOT better. Thanks again!